_adr/90/index.adoc

```diff
@@ -0,0 +1,124 @@
+Instead a secret is created and only references to the secrets will be added before the ManagedKafka is created in the data plane.
+
+### Threat model
+
+Once the certificate are in the data plane in form of a Kubernetes secret, a privileged users can still see them. 
+If this is on a enterprise data plane cluster without good security standards/policy, then anyone with the private key can impersonate the server or eavesdrop on a connection with the server. 
+For non enterprise data plane clusters, this is mitigated by only allowing SREs to have the privilege / permissions to see these secrets.
+For enterprise data plane clusters, customers will be advised to do so as well.
+Morever, since there will be a different certificate per kafka instance, if one certificate is compromised, it won't affect others and it can be safely revoked. 
+The proposed solution also limits the possibility of an owner of a enterprise data plane cluster, eavesdropping on another data plane cluster that they do not own as the certificate keys will be different. 
+
+NOTE: If the shared certificate for an existing Kafka is to be compromised, then it opens up a securiry risk on all existing Kafkas. However, the chances of this security breach occurring is mitigated by first and foremost ensuring only SREs have privileged access to data plane clusters. And second, if a the security issue is detected then certificates can be easily revoked and new ones autocreated 
+
+## Alternatives Considered / Rejected
+
+1. Deploying https://www.redhat.com/sysadmin/cert-manager-operator-openshift[cert-manager operator] to the data plane. Rejected to avoid exposing domain registrar credentials and to avoid adding another component that consumes data plane resources and that needs to be monitored.
+
+2. Migrating existing their Kafkas. This change is disruptive to existing customers hence why it was rejected. 
+The Kafka fleet manager will thus ensure that existing Kafkas continue to operate using a shared wildcard certificate on the `*.kafka.bf2.dev` domain i.e the proposed solution will only be applied to new Kafkas.
+
+3. Deploying https://www.redhat.com/sysadmin/cert-manager-operator-openshift[cert-manager operator] in the Control Plane. 
+The Control plane then handles the pushing of the Certificate request CR, watching for the created secrets and storing them securely. 
+Rejected because this is technically complex and has no clear benefit over the propsed solution of using a library.
+It also violates the design intentions of the fleet manager by turning into an Kubernetes operator. 
+Finally, it's another component deployed that has to be monitored and managed in the control plane which forces every one (especially dev environments) to have a running Kubernetes cluster running during development
+
+## Challenges
+
+1. The drawback is that the solution can't or won't handle the migration of already existing Kafkas.
```

Comment:
while it is covered in the alternative. A challenge I see here is that we are effectively turning the fleet manager into a single ACME provider cert-manager and so are likely to run into bugs and issues that cert-manager has nailed down over time

Reply:
The long term is that HCG might act as cert broker, in the short term and to make it clear, the fleet manager doesn't implement this logic. We reuse existing ACME libraries - and leaving the cert management to the ACME library.